src/message_filter.rs

```diff
@@ -56,92 +62,60 @@ impl<Message: Hash> MessageFilter<Message> {
     pub fn insert(&mut self, message: &Message) -> usize {
         self.remove_expired();
         let hash_code = hash(message);
-        if let Some(index) = self.entries.iter().position(|t| t.hash_code == hash_code) {
-            let mut timestamped_message = self.entries.remove(index);
-            timestamped_message.update_expiry_point(self.time_to_live);
-            let count = timestamped_message.increment_count();
-            self.entries.push(timestamped_message);
-            count
-        } else {
-            self.entries.push(TimestampedMessage::new(hash_code, self.time_to_live));
-            1
+        let now = Instant::now();
+        self.timeout_queue.push_back((hash_code, now));
```

Comment:
I think we only ever use this time to check expiry? In which case it's probably better to insert `now + time_to_live`.